x-pack/legacy/plugins/security/server/routes/views/login.js

```diff
@@ -8,9 +8,8 @@ import { get } from 'lodash';
 
 import { parseNext } from '../../lib/parse_next';
 
-export function initLoginView(server, xpackMainPlugin) {
+export function initLoginView({ config: { cookieName } }, server, xpackMainPlugin) {
```

Comment:
We definitely should somehow improve that (maybe with something like Hapi's `try` auth mode? we should prevent redirects during authentication for such mode though). But the route itself opts out of authentication and hence `isAuthenticated` will return `false`

Reply:
> But the route itself opts out of authentication and hence isAuthenticated will return false

yeah. that's kinda expected...because we don't run interceptor we cannot say for sure whether cookies aren't stale.

Reply:
>  because we don't run interceptor we cannot say for sure whether cookies aren't stale.

It has more problems than that :) Some providers (e.g. SAML) relies on intermediate cookies that used during SAML handshake, so if start handshake you can no longer get back to login page. That's something I'm going to solve soon, but not in this PR.

Reply:
created https://github.com/elastic/kibana/issues/41959